src/main/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotDao.java

```diff
@@ -18,11 +19,12 @@ public TSnePlotDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
 
     private static final String SELECT_T_SNE_PLOT_WITH_EXPRESSION_STATEMENT =
             "SELECT tsne.cell_id, tsne.x, tsne.y, analytics.expression_level " +
-            "FROM scxa_tsne AS tsne " +
-                "LEFT JOIN " +
-                "(SELECT * FROM scxa_analytics WHERE gene_id=:gene_id) AS analytics " +
-                "ON analytics.cell_id=tsne.cell_id AND analytics.experiment_accession=tsne.experiment_accession " +
-            "WHERE tsne.experiment_accession=:experiment_accession AND tsne.perplexity=:perplexity";
+                    "FROM scxa_tsne AS tsne " +
+                    "LEFT JOIN " +
+                    "(SELECT * FROM scxa_analytics WHERE gene_id=:gene_id) AS analytics " +
```

Comment:
I would suggest here that the experiment accession is set inside the sub query, and not outside. That means that the subquery uses its index and that it returns less results, that are otherwise more costly to filter out later in the result of the subquery (which by definition, won't have an index).

Reply:
Otherwise, this could be costly for genes in organisms that have a lot of cells/experiments.

Reply:
Yes, please change this, doing an EXPLAIN on these two queries:

```
SELECT * FROM scxa_analytics WHERE gene_id = 'ENSG00000019582'
```

and

```
SELECT * FROM scxa_analytics WHERE gene_id = 'ENSG00000019582' AND experiment_accession = 'E-MTAB-6678';
```

the cost of the first is 1024803, much larger compared to the cost on the second 1473.27!!!

```

Reply:
(the changes are suggested in the comment below).

Reply:
Implemented.